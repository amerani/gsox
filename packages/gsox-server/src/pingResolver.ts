import { pubSub } from "./pubSubProvider";
import { IResolvers, IResolverObject } from "../node_modules/graphql-tools";

const pingSubscribe: IResolverObject = {
      subscribe() {
            setInterval(() => pubSub.publish("test", {
                Ping: { id: 0 },
            }), 2000);
            return pubSub.asyncIterator("test");
      },
};

const pingResolver: IResolvers<any, any> = {
      Ping: pingSubscribe
}

export default pingResolver;