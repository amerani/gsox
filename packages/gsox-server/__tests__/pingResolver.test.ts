import pingResolver from "../src/pingResolver";

test('should create server', () => {
      expect(pingResolver).toMatchSnapshot();
})