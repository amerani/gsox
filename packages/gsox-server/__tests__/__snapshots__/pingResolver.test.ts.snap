// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create server 1`] = `
Object {
  "Ping": Object {
    "subscribe": [Function],
  },
}
`;
